app/tests/spec/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const {assert} = require('chai');
+  const RecoveryKeys = require('lib/crypto/recovery-keys');
+  const validate = require('../../../../scripts/lib/validate');
+
+  const uid = 'aaaaabbbbbcccccdddddeeeeefffff00';
+  const recoveryKey = 'AFAUEQ2EIVDEOSCJJJFUYTKOJ5IFCUST';
+  const recoveryData = {
+    kB: '000000111111222222333333444444555555666666777777888888999999ABCD'
+  };
+  const expectedRecoveryKeyId = 'edc243a821582ee9e979583be9989ee7';
+  const expectedRecoveryEnc = '6737e3c3c678f0dc67abfb853115e7c702933795a3662a99d1b27b6e645dc506';
+  // const expectedBundle = 'eyJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiZGlyIiwia2lkIjoiODE4NDIwZjBkYTU4ZDIwZjZhZTR' +
+  //   'kMmM5YmVhYjkyNTEifQ..D29EXHp8ubLvftaZ.xHJd2Nl2Uco2RyywYPLkUU7fHpgO2FztY12Zjpq1ffiyLRIUcQVfmiNC6aMiHB' +
+  //   'l7Hp-lXEbb5mR1uXHrTH9iRXEBVaAfyf9KEAWOukWGVSH8EaOkr7cfu2Yr0K93Ec8glsssjiKp8NGB8VKTUJ-lmBv2cIrG68V4eTUVDo' +
+  //   'DhMbXhrF-Mv4JNeh338pPeatTnyg.Ow2bhEYWxzxfSPMxVwKmSA';
+  // const encryptionOptions = {
+  //   iv: 'eeddccbbaa99887766554433'
+  // };
+
+  describe('lib/crypto/recovery-keys', () => {
+    describe('generateRecoveryKey', () => {
+      it('should generate base32 string', () => {
+        return RecoveryKeys.generateRecoveryKey(1000)
```

Comment:
I understand this test, but it seems like it could feasibly pass one run and not the next, which would be maddening and a total PITA to debug.